cdap-security-spi/src/main/java/co/cask/cdap/security/spi/authentication/SecurityRequestContext.java

```diff
@@ -64,9 +65,26 @@ public static void setUserIP(String userIPParam) {
   }
 
   /**
+   * @return the principal type set on the current thread or PrincipalType.USER if not set
```

Comment:
Why do we need this ? Doesn't the SecurityRequestContext represent the logged in user ? Type is  always user no ?